tests/test_datasets/test_loaders.py

```diff
@@ -0,0 +1,253 @@
+    assert find_dataset_path(name, ext=".npz", raises=False) is not None, "no .npz in dataset"
+
+    n_files = len(data.contents())
+    assert n_files == 4 or n_files == 5, "not enough files in dataset"
+    assert len(data.README) > 0, "readme contains no data"
+    assert len(data.meta) > 0, "metadata is empty"
+
+    if n_files == 5:
+        assert len(data.citation) > 0, "citation.bib is empty"
+
+    assert "features" in data.meta, "no features in metadata"
+    assert "target" in data.meta, "no target in metadata"
+
+
+def assert_valid_corpus(corpus, name):
+    __tracebackhide__ = True
+    assert isinstance(corpus, Corpus), "not a Corpus object"
+    assert name in DATASETS, "corpus not in manifest"
+
+    assert dataset_exists(name), "corpus directory does not exist"
+    assert dataset_archive(name, DATASETS[name]["signature"]), "corpus archive does not match signature"
+
+    n_contents = len(corpus.contents())
+    assert n_contents > 2, "not enough files/directories in corpus"
+    assert len(corpus.README) > 0, "readme contains no data"
+    assert corpus.citation is None or len(corpus.citation) > 0, "citation.bib is empty"
+
+
+def assert_valid_pandas(data):
+    __tracebackhide__ = True
+    # Get raw data frame
+    df = data.to_dataframe()
+    assert isinstance(df, pd.DataFrame), "raw dataframe is wrong type"
+
+    # Get pandas data
+    X, y = data.to_pandas()
+    assert isinstance(X, pd.DataFrame), "X is not a DataFrame"
+    assert isinstance(y, pd.Series), "y is not a Series"
+
+    # Assert pandas is returned from to_data()
+    X, y = data.to_data()
+    assert isinstance(X, pd.DataFrame), "to_data does not return pandas"
+    assert isinstance(y, pd.Series), "to_data does not return pandas"
+
+
+def assert_valid_numpy(data):
+    __tracebackhide__ = True
+    X, y = data.to_numpy()
+    assert isinstance(X, np.ndarray), "X is not a numpy array"
+    assert isinstance(y, np.ndarray), "y is not a numpy array"
+    assert X.ndim == 2 and y.ndim == 1, "X and y dimensions are incorrect"
+
+    # Patch pandas and make defaults assertions
+    X, y = data.to_data()
+    assert isinstance(X, np.ndarray), "to_data does not return numpy"
+    assert isinstance(y, np.ndarray), "to_data does not return numpy"
+
+    with pytest.raises(DatasetsError):
+        data.to_pandas(), "exception not raised when pandas unavailable"
+
+
+##########################################################################
+## Test Cases
+##########################################################################
+
+class TestDatasetLoaders(object):
+    """
+    Test the dataset loading functions
+    """
+
+    def test_load_concrete(self):
+        """
+        Test loading the concrete regression dataset
+        """
+        data = load_concrete(return_dataset=True)
+        assert_valid_dataset(data, "concrete")
+
+        X, y = load_concrete()
+        assert X.shape == (1030, 8)
```

Comment:
Will fail if dataset versions have changed unexpectedly, causing shape to change.